mmcv/cnn/bricks/transformer.py

```diff
@@ -0,0 +1,599 @@
+        dropout (float): A Dropout layer on attn_output_weights. Default 0.0.
+    """
+
+    def __init__(self, feat_channels, num_heads, dropout=0.0):
+        super(MultiheadAttention, self).__init__()
+        assert feat_channels % num_heads == 0, 'feat_channels must be ' \
+            f'divisible by num_heads. got {feat_channels} and {num_heads}.'
+        self.feat_channels = feat_channels
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.attn = nn.MultiheadAttention(feat_channels, num_heads, dropout)
+        self.dropout = nn.Dropout(dropout)
+
+    def forward(self,
+                x,
+                key=None,
+                value=None,
+                residual=None,
+                query_pos=None,
+                key_pos=None,
+                attn_mask=None,
+                key_padding_mask=None):
+        """Forward function for MultiheadAttention."""
+        query = x
+        if key is None:
+            key = query
+        if value is None:
+            value = key
+        if residual is None:
+            residual = x
+        if key_pos is None:
+            if query_pos is not None and key is not None:
+                if query_pos.shape == key.shape:
+                    key_pos = query_pos
+        if query_pos is not None:
+            query += query_pos
+        if key_pos is not None:
+            key += key_pos
+        out = self.attn(
+            query,
+            key,
+            value=value,
+            attn_mask=attn_mask,
+            key_padding_mask=key_padding_mask)[0]
+        return residual + self.dropout(out)
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f'(feat_channels={self.feat_channels}, '
+        repr_str += f'num_heads={self.num_heads}, '
+        repr_str += f'dropout={self.dropout})'
+        return repr_str
+
+
+class FFN(nn.Module):
+    """Implements feed-forward networks (FFNs) with residual connection.
+
+    Args:
+        feat_channels (int): The feature dimension. Same as
+            `MultiheadAttention`.
+        feedforward_channels (int): The hidden dimension of FFNs.
+        num_fcs (int): The number of fully-connected layers in FFNs.
+        act_cfg (dict): The activation config for FFNs.
+        dropout (float): Probability of an element to be zeroed. Default 0.0.
+    """
+
+    def __init__(self,
+                 feat_channels,
+                 feedforward_channels=2048,
+                 num_fcs=2,
+                 act_cfg=dict(type='ReLU', inplace=True),
+                 dropout=0.0):
+        super(FFN, self).__init__()
+        assert num_fcs >= 2, 'num_fcs should be no less ' \
+            f'than 2. got {num_fcs}.'
+        self.feat_channels = feat_channels
+        self.feedforward_channels = feedforward_channels
+        self.num_fcs = num_fcs
+        self.act_cfg = act_cfg
+        self.dropout = dropout
+        # NOTE only when act_cfg['type'] not in ['Tanh', 'PReLU',
```

Comment:
These comments should not be put here. The users should be aware of that in the documentation of `build_activation_layer` or some place else.